academic-observatory-workflows/academic_observatory_workflows/crossref_metadata_telescope/telescope.py

```diff
@@ -0,0 +1,326 @@
+    retries: int = 3
+    gke_image = "us-docker.pkg.dev/academic-observatory/academic-observatory/academic-observatory:latest"
+    gke_namespace: str = "coki-astro"
+    gke_startup_timeout_seconds: int = 300
+    gke_conn_id: str = "gke_cluster"
+    docker_astro_uid: int = 50000
+    gke_volume_name: str = "crossref-metadata"
+    gke_volume_path: str = "/data"
+    gke_zone: str = "us-central1"
+    gke_volume_size: int = 2500
+    gke_resource_overrides: Optional[dict] = None
+
+
+def create_dag(dag_params: DagParams) -> DAG:
+    """The Crossref Metadata DAG"""
+
+    task_resources = _TaskResources(overrides=dag_params.gke_resource_overrides)
+    # Common @task.kubernetes params
+    volume_mounts = [k8s.V1VolumeMount(mount_path=dag_params.gke_volume_path, name=dag_params.gke_volume_name)]
+    volumes = [
+        k8s.V1Volume(
+            name=dag_params.gke_volume_name,
+            persistent_volume_claim=k8s.V1PersistentVolumeClaimVolumeSource(claim_name=dag_params.gke_volume_name),
+        )
+    ]
+    kubernetes_task_params = dict(
+        image=dag_params.gke_image,
+        security_context=k8s.V1PodSecurityContext(
+            fs_group=dag_params.docker_astro_uid,
+            fs_group_change_policy="OnRootMismatch",
+            run_as_group=dag_params.docker_astro_uid,
+            run_as_user=dag_params.docker_astro_uid,
+        ),
+        do_xcom_push=True,
+        get_logs=True,
+        in_cluster=False,
+        kubernetes_conn_id=dag_params.gke_conn_id,
+        log_events_on_failure=True,
+        namespace=dag_params.gke_namespace,
+        startup_timeout_seconds=dag_params.gke_startup_timeout_seconds,
+        env_vars={"DATA_PATH": dag_params.gke_volume_path},
+        volumes=volumes,
+        volume_mounts=volume_mounts,
+    )
+
+    @dag(
+        dag_id=dag_params.dag_id,
+        start_date=dag_params.start_date,
+        schedule=dag_params.schedule,
+        catchup=dag_params.catchup,
+        max_active_runs=dag_params.max_active_runs,
+        tags=["academic-observatory"],
+        default_args={
+            "owner": "airflow",
+            "on_failure_callback": on_failure_callback,
+            "retries": dag_params.retries,
+            "queue": dag_params.queue,
+        },
+    )
+    def crossref_metadata():
+        @task
+        def _fetch_release(dag_params: DagParams, **context) -> dict:
+            """Fetch the release for this month, making sure that it exists."""
+            from academic_observatory_workflows.crossref_metadata_telescope.tasks import fetch_release
+
+            return fetch_release(
+                run_id=context["run_id"],
+                data_interval_start=context["data_interval_start"],
+                data_interval_end=context["data_interval_end"],
+                cloud_workspace=dag_params.cloud_workspace,
+                crossref_metadata_conn_id=dag_params.crossref_metadata_conn_id,
+                dag_id=dag_params.dag_id,
+            )
```

Comment:
Do these need to start with an underscore? It is cleaner without them.

I think the Kubernetes based task functions and functions within them can have the same name? 

For tasks like these that run as Python operators, so that there is no conflict, and so that there are less imports, you could instead import the tasks module once globally, alias it and then reuse it:

```python
from academic_observatory_workflows.crossref_metadata_telescope.tasks as tasks

@task
def fetch_release(dag_params: DagParams, **context) -> dict:
    """Fetch the release for this month, making sure that it exists."""

    return tasks.fetch_release(
        run_id=context["run_id"],
        data_interval_start=context["data_interval_start"],
        data_interval_end=context["data_interval_end"],
        cloud_workspace=dag_params.cloud_workspace,
        crossref_metadata_conn_id=dag_params.crossref_metadata_conn_id,
        dag_id=dag_params.dag_id,
    )
```